src/Broadway/CommandHandling/ClosureCommandHandler.php

```diff
@@ -31,13 +32,22 @@ public function add(\Closure $handler): void
         $reflection = new \ReflectionFunction($handler);
         $reflectionParams = $reflection->getParameters();
 
-        if (!isset($reflectionParams[0]) || !$reflectionParams[0]->getClass()) {
-            throw new ClosureParameterNotAnObjectException();
+        $reflectionType = $reflectionParams[0]->getType();
+
+        $name = null;
+        if ($reflectionType instanceof \ReflectionNamedType) {
+            /** @var \ReflectionNamedType $reflectionNamed */
+            $reflectionNamed = $reflectionParams[0]->getType();
+            if (!$reflectionNamed->isBuiltin()) {
+                $name = new ReflectionClass($reflectionNamed->getName());
+            }
         }
 
-        $index = $reflectionParams[0]->getClass()->getName();
+        if (!isset($reflectionParams[0]) || !$name) {
```

Comment:
no need to check for `!isset($reflectionParams[0]` because on line 35 you already assume it is set.

Reply:
Removed